wasmtime-jit/src/compiler.rs

```diff
@@ -151,6 +151,18 @@ impl Compiler {
         })
     }
 
+    /// Create and publish a trampoline for invoking a function.
```

Comment:
Could you add a comment to `get_trampoline` mentioning that the code it returns needs to be published (with `publish_compiled_code`) before it can be executed, and pointing to `get_published_trampoline` as a way to have this done automatically?

And, we should mention that calling `get_published_trampoline` many times will use memory inefficiently, because publishing between each trampoline creation means we'll use a whole page for each trampoline. When feasible, it's preferable to create all the trampolines (with `get_trampoline`) first, and then call `publish_compiled_code` once.

Reply:
Actually, looking at this more, `get_trampoline` and `publish_compiled_code` are not public functions, so these comments wouldn't immediately make sense. I think this is a symptom of `cranelift-jit` not being entirely clear about what level of API it wants to be. Hopefully we'll sort all this out with the new APIs.